scipy/integrate/odepack.py

```diff
@@ -257,3 +270,31 @@ def odeint(func, y0, t, args=(), Dfun=None, col_deriv=0, full_output=0,
         return output[0]
     else:
         return output
+
+
+def _make_empty_info(n):
+    """
+    Create an `info` dictionary for the return value of odeint in the
+    case where y0 has length 0 and full_output is True.
+    """
+    nm1 = n - 1
+    info = {
+        'hu': np.zeros(nm1, dtype=np.float64),
+        'tcur': np.zeros(nm1, dtype=np.float64),
+        'tolsf': np.zeros(nm1, dtype=np.float64),
+        'tsw': np.zeros(nm1, dtype=np.float64),
+        'nst': np.zeros(nm1, dtype=np.int32),
+        'nfe': np.zeros(nm1, dtype=np.int32),
+        'nje': np.zeros(nm1, dtype=np.int32),
+        'nqu': np.zeros(nm1, dtype=np.int32),
+        'imxer': -1,
+        'lenrw': 0,
+        'leniw': 0,
+        'mused': np.zeros(nm1, dtype=np.int32),
+    }
+    if nm1 == 0:
+        info['message'] = 'Nothing was done; the integration time was 0.'
+    else:
+        info['message'] = 'Integration successful.'
```

Comment:
Isn't nothing done even in this case?

Reply:
I'm preserving the invariants where `len(t) == 1` gives the message `'Nothing was done...'` and `len(t) > 1` gives the message `'Integration successful.'`.  I don't think even this distinction is useful, and I'd be happy to always give a simple "success" message in any case that is not an error, but I'm copying the current behavior for backwards compatibility.